tap_appsflyer/__init__.py

```diff
@@ -378,37 +378,40 @@ def sync_in_app_events():
         "original_url",
     )
 
+    stop_time = datetime.datetime.now()
     from_datetime = get_start("in_app_events")
-    to_datetime = get_stop(from_datetime)
-
-    if to_datetime < from_datetime:
-        LOGGER.error("to_datetime (%s) is less than from_endtime (%s).", to_datetime, from_datetime)
-        return
-
-    params = dict()
-    params["from"] = from_datetime.strftime("%Y-%m-%d %H:%M")
-    params["to"] = to_datetime.strftime("%Y-%m-%d %H:%M")
-    params["api_token"] = CONFIG["api_token"]
-
-    url = get_url("in_app_events", app_id=CONFIG["app_id"])
-    request_data = request(url, params)
-
-    csv_data = RequestToCsvAdapter(request_data)
-    reader = csv.DictReader(csv_data, fieldnames)
-
-    next(reader) # Skip the heading row
-
-    bookmark = from_datetime
-    for i, row in enumerate(reader):
-        record = xform(row, schema)
-        singer.write_record("in_app_events", record)
-        # AppsFlyer returns records in order of most recent first.
-        if utils.strptime(record["event_time"]) > bookmark:
-            bookmark = utils.strptime(record["event_time"])
-
-    # Write out state
-    utils.update_state(STATE, "in_app_events", bookmark)
-    singer.write_state(STATE)
+    to_datetime = get_stop(from_datetime, stop_time, 10)
+
+    while to_datetime <= stop_time:
+        LOGGER.info("Syncing data from %s to %s", from_datetime, to_datetime)
+        params = dict()
+        params["from"] = from_datetime.strftime("%Y-%m-%d %H:%M")
+        params["to"] = to_datetime.strftime("%Y-%m-%d %H:%M")
+        params["api_token"] = CONFIG["api_token"]
+
+        url = get_url("in_app_events", app_id=CONFIG["app_id"])
+        request_data = request(url, params)
+
+        csv_data = RequestToCsvAdapter(request_data)
+        reader = csv.DictReader(csv_data, fieldnames)
+
+        next(reader) # Skip the heading row
+
+        bookmark = from_datetime
+        for i, row in enumerate(reader):
+            record = xform(row, schema)
+            singer.write_record("in_app_events", record)
+            # AppsFlyer returns records in order of most recent first.
+            if utils.strptime(record["event_time"]) > bookmark:
+                bookmark = utils.strptime(record["event_time"])
+
+        # Write out state
+        utils.update_state(STATE, "in_app_events", bookmark)
+        singer.write_state(STATE)
+
+        # Move the timings forward
+        from_datetime = to_datetime
+        to_datetime = get_stop(from_datetime, stop_time)
```

Comment:
Should this be passing 10 as well? Or do we just want to do two rounds through the timespan?

Reply:
I was just going to keep the default 30 here as it wasn't causing an issue.